gateway-sp-comms/src/single_sp.rs

```diff
@@ -1002,6 +1076,21 @@ impl SingleSp {
         result.result.and_then(expect_caboose_value)
     }
 
+    pub async fn read_component_caboose_string(
+        &self,
+        component: SpComponent,
+        slot: u16,
+        key: [u8; 4],
+    ) -> Result<String> {
+        let value = self.read_component_caboose(component, slot, key).await?;
+
+        Ok(if value.is_ascii() {
+            String::from_utf8(value).unwrap()
+        } else {
+            hex::encode(value)
+        })
+    }
```

Comment:
Do we need a crate version bump for this addition?

Reply:
I'm not sure why, can you say more?  This is just a helper function to consolidate the "read, then convert to string or hex" pattern from a few places.

Reply:
Ah I think I misread. I thought it was a change to `read_component_caboose`  as opposed to a helper function. You are right, we shouldn't need a bump.

Reply:
(we've also been playing fast and loose with semver in this crate, since we control when we bump the version in Hubris's `Cargo.lock`)